src/agent/doc/index.rst

```diff
@@ -223,6 +223,37 @@ to the LLM::
 
     $agent = new Agent($platform, $model, inputProcessor: [$toolProcessor], outputProcessor: [$toolProcessor]);
 
+If you want to expose the underlying error to the LLM, you can throw a custom exception that implements `ToolExecutionExceptionInterface`::
+
+    use Symfony\AI\Agent\Toolbox\Exception\ToolExecutionExceptionInterface;
+
+    class EntityNotFoundException extends \RuntimeException implements ToolExecutionExceptionInterface
+    {
+        public function __construct(private string $entityName, private int $id)
+        {
+        }
+
+        public function getToolCallResult(): mixed
+        {
+            return \sprintf('No %s found with id %d', $this->entityName, $this->id);
+        }
+    }
+
+    #[AsTool('get_user_age', 'Get age by user id')]
```

Comment:
not sure if possible, but wouldn't it be better if you would be able to provide this info via callable in the attribute to not couple you code with the 3rd party exception?

```suggestion
    #[AsTool('get_user_age', 'Get age by user id', exceptions: [EntityNotFoundException::class => static fn(THE TOOL CALL ARGUMENTS HERE)....])]
```

Or is this not possible? Or is it a bad idea?